src/GafferScene/CameraQuery.cpp

```diff
@@ -0,0 +1,551 @@
+					parameters->writable()[g_shutter] = new CompoundData( {
+						{ g_source, shutterSource },
+						{ g_value, shutterValue }
+					} );
+
+					// Fall back to default values for all other registered camera parameters.
+					for( const auto &target : Metadata::targetsWithMetadata( "parameter:camera:*", g_defaultValue ) )
+					{
+						const std::string name = target.string().substr( 17 );
+						if( !parameters->readable().count( name ) )
+						{
+							parameters->writable()[name] = new CompoundData( {
+								{ g_source, g_sourceFallback },
+								{ g_value, Metadata::value( target, g_defaultValue )->copy() }
+							} );
+						}
+					}
+
+					// Create virtual parameters, for convenience these are pre-computed
+					// from the camera so they can be later accessed like a regular parameter.
+					const Imath::V2f aperture = cameraWithGlobals->getAperture();
+					parameters->writable()["apertureAspectRatio"] = new CompoundData( {
+						{ g_source, g_sourceCamera },
+						{ g_value, new FloatData( aperture[0] / aperture[1] ) }
+					} );
+					parameters->writable()["fieldOfView"] = new CompoundData( {
+						{ g_source, g_sourceCamera },
+						{ g_value, new V2fData( cameraWithGlobals->calculateFieldOfView() ) }
+					} );
+					parameters->writable()["frustum"] = new CompoundData( {
+						{ g_source, g_sourceCamera },
+						{ g_value, new Box2fData( cameraWithGlobals->frustum() ) }
+					} );
+				}
+			}
+		}
+
+		static_cast<AtomicCompoundDataPlug *>( output )->setValue( parameters );
+		return;
+	}
+	else if( outPlug()->isAncestorOf( output ) )
+	{
+		const ValuePlug *oPlug = outPlug( output );
+
+		const IECore::ConstCompoundDataPtr parameters = internalParametersPlug()->getValue();
+		assert( parameters );
+
+		if( output == oPlug->getChild( g_sourcePlugIndex ) )
+		{
+			const std::string parameterName = queryPlug( output )->getValue();
+			std::string source = g_sourceNone->readable();
+			if( parameterName.size() )
+			{
+				CompoundDataMap::const_iterator it = parameters->readable().find( parameterName );
+				if( it != parameters->readable().end() )
+				{
+					if( const auto c = runTimeCast<CompoundData>( it->second.get() ) )
+					{
+						if( const auto s = runTimeCast<StringData>( c->member( g_source ) ) )
+						{
+							source = s->readable();
+						}
+					}
+				}
+			}
+
+			static_cast<StringPlug *>( output )->setValue( source );
+			return;
+		}
+
+		const ValuePlug *valuePlug = oPlug->getChild<ValuePlug>( g_valuePlugIndex );
+		if( output == valuePlug || valuePlug->isAncestorOf( output ) )
+		{
+			const StringPlug *childQueryPlug = queryPlug( output );
+
+			const std::string parameterName = childQueryPlug->getValue();
+			ConstObjectPtr object;
+			if( parameterName.size() )
+			{
+				CompoundDataMap::const_iterator it = parameters->readable().find( parameterName );
+				if( it != parameters->readable().end() )
+				{
+					if( const auto c = runTimeCast<CompoundData>( it->second.get() ) )
```

Comment:
I think we could call `member()` here instead. Maybe we can lift all of this out of the `source/value` conditionals and initialise `const CompoundData *parameterData` before we hit them.